### PR TITLE
`requiresig` plugin for Prow.

### PR DESCRIPTION
**Pull request by cjwagner**
This replace the `sig-mention` munger, but there is already a `sig-mention` plugin that reiterates sig mentions from non-members so I named this plugin `requiresig` since it is the component that actually enforces the requirement that issues have a sig label.
ref #3331

### CODE REVIEW

**Comment by BenTheElder** [comment]
"Requires IG"? /s
Naming is hard :joy:

**Comment by cjwagner** [comment]
> "Requires IG"? /s

"Require SIG"? 😛

**Comment by BenTheElder** [comment]
Maybe it needs to be `require-sig`  🎨 🖌 ➡️  :bike: 🏚

File: prow/plugins/plugins.go [hunk 1]
```diff
@@ -214,6 +215,10 @@ func (pa *PluginAgent) MDYAMLEnabled(org, repo string) bool {
 
 }
 
+type RequireSIG struct {
+	GroupListURL string `json:"group_list_url,omitempty"`
```
Comment (kargakis) [inline prow/plugins/plugins.go]: godoc what this is about

File: prow/plugins/requiresig/requiresig.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+	}
+	return false
+}
+
+// handle is the workhorse notifying issue owner to add a sig label if there is none
+// The algorithm:
+// (1) return if this is not an opened, labelled, or unlabelled event or if the issue is closed.
+// (2) find if the issue has a sig label
+// (3) find if the issue has a needs-sig label
+// (4) if the issue has both the sig and needs-sig labels, remove the needs-sig label and comment.
```
Comment (kargakis) [inline prow/plugins/requiresig/requiresig.go]: Do we need to comment here?

Comment (cjwagner) [reply prow/plugins/requiresig/requiresig.go]: I meant "remove both the needs-sig label and the comment".  I'll reword this to be more clear.

**Comment by kargakis** [comment]
> Maybe it needs to be require-sig

There is also `milestonestatus` and `config_updater` :) 🖌 🖌 🖌

**Comment by cjwagner** [comment]
Comments addressed. PTAL.
I changed `requiresig` to `require-sig` where possible.

File: prow/github/types.go [hunk 3]
```diff
@@ -203,6 +203,7 @@ type IssueEvent struct {
 	Action IssueEventAction `json:"action"`
 	Issue  Issue            `json:"issue"`
 	Repo   Repo             `json:"repository"`
+	Label  Label            `json:"label"`
```
Comment (kargakis) [inline prow/github/types.go]: Where is this used?

Comment (cjwagner) [reply prow/github/types.go]: It is not used anymore, I switched to just using `IssueEvent.Issue.Labels`.  I removed the commit that added this.

**Comment by BenTheElder** [comment]
xref: https://github.com/kubernetes/test-infra/issues/3331, [mungegithub is flakey](https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/test-infra/5662/pull-test-infra-bazel/10509/)
/retest
